waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/server/FederatedHMSHandler.java

```diff
@@ -1369,19 +1377,48 @@ public List<String> set_ugi(String user_name, List<String> group_names) throws M
   @Loggable(value = Loggable.DEBUG, skipResult = true, name = INVOCATION_LOG_NAME)
   public String get_delegation_token(String token_owner, String renewer_kerberos_principal_name)
       throws MetaException, TException {
-    return getPrimaryClient().get_delegation_token(token_owner, renewer_kerberos_principal_name);
+    try {
+      return MetaStoreProxyServer.getSaslServerAndMDT().getDelegationTokenManager()
```

Comment:
remove these static calls the DelegationTokenManager should be a field inject in the constructor of this class.

Reply:
I started with injection, but then there will be cyclic dependencies. federatehmshandler ---> metastoreproxyserver --->others ------> federatehmshandler.
I've been thinking for a long time but haven't figured out how to solve it.It may be necessary to introduce third-party classes to solve this problem